File: solutions/ruby/resistor-color-duo/1/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

module ResistorColorDuo
  def self.value(colors)
    "#{to_val(colors[0])}#{to_val(colors[1])}".to_i
  end

  private

  def self.to_val(color)
    case color
    when 'black'
      0
    when 'brown'
      1
    when 'red'
      2
    when 'orange'
      3
    when 'yellow'
      4
    when 'green'
      5
    when 'blue'
      6
    when 'violet'
      7
    when 'grey'
      8
    when 'white'
      9
    end
  end
end

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(colors: string[]): number {
  const result = colors.slice(0, 2).reduce(
    (carry: string, value: string) => carry + colorCode(value).toString(),
    ''
  );

  return Number(result);
}

export const colorCode = (color: string): number => {
  return COLORS.findIndex(v => v === color);
}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];